components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -468,197 +416,159 @@ void AdsServiceImpl::InitializePrefChangeRegistrar() {
+        wallet->payment_id, base::Base64Encode(wallet->recovery_seed));
   }
+}
+
+void AdsServiceImpl::Initialize() {
+  InitializeDatabase();
 
-  last_bat_ads_service_restart_time_ = base::Time::Now();
+  BackgroundHelper::GetInstance()->AddObserver(this);
+
+  g_brave_browser_process->resource_component()->AddObserver(this);
+
+  RegisterResourceComponentsForDefaultLocale();
 
-  return delay;
+  GetDeviceId();
 }
 
-void AdsServiceImpl::Start(const uint32_t number_of_start) {
-  GetDeviceId(number_of_start);
+void AdsServiceImpl::InitializeDatabase() {
+  DCHECK(!database_);
+
+  database_ = std::make_unique<ads::Database>(
+      base_path_.AppendASCII("database.sqlite"));
 }
 
-void AdsServiceImpl::GetDeviceId(const uint32_t number_of_start) {
-  device_id_->GetDeviceId(base::BindOnce(&AdsServiceImpl::OnGetDeviceId,
-                                         AsWeakPtr(), number_of_start));
+void AdsServiceImpl::GetDeviceId() {
+  device_id_->GetDeviceId(
+      base::BindOnce(&AdsServiceImpl::OnGetDeviceId, AsWeakPtr()));
 }
 
-void AdsServiceImpl::OnGetDeviceId(const uint32_t number_of_start,
-                                   std::string device_id) {
+void AdsServiceImpl::OnGetDeviceId(std::string device_id) {
   sys_info_.device_id = std::move(device_id);
 
-  DetectUncertainFuture(number_of_start);
+  DetectUncertainFuture();
 }
 
-void AdsServiceImpl::DetectUncertainFuture(const uint32_t number_of_start) {
-  brave_rpill::DetectUncertainFuture(base::BindOnce(
-      &AdsServiceImpl::OnDetectUncertainFuture, AsWeakPtr(), number_of_start));
+void AdsServiceImpl::DetectUncertainFuture() {
+  brave_rpill::DetectUncertainFuture(
+      base::BindOnce(&AdsServiceImpl::OnDetectUncertainFuture, AsWeakPtr()));
 }
 
-void AdsServiceImpl::OnDetectUncertainFuture(const uint32_t number_of_start,
-                                             const bool is_uncertain_future) {
+void AdsServiceImpl::OnDetectUncertainFuture(const bool is_uncertain_future) {
   sys_info_.is_uncertain_future = is_uncertain_future;
 
-  EnsureBaseDirectoryExists(number_of_start);
+  EnsureBaseDirectoryExists();
 }
 
-void AdsServiceImpl::EnsureBaseDirectoryExists(const uint32_t number_of_start) {
+void AdsServiceImpl::EnsureBaseDirectoryExists() {
   base::PostTaskAndReplyWithResult(
       file_task_runner_.get(), FROM_HERE,
       base::BindOnce(&EnsureBaseDirectoryExistsOnFileTaskRunner, base_path_),
-      base::BindOnce(&AdsServiceImpl::OnEnsureBaseDirectoryExists, AsWeakPtr(),
-                     number_of_start));
+      base::BindOnce(&AdsServiceImpl::OnEnsureBaseDirectoryExists,
+                     AsWeakPtr()));
 }
 
-void AdsServiceImpl::OnEnsureBaseDirectoryExists(const uint32_t number_of_start,
-                                                 const bool success) {
+void AdsServiceImpl::OnEnsureBaseDirectoryExists(const bool success) {
   if (!success) {
     VLOG(0) << "Failed to create base directory";
     return;
```

Comment:
bat ads will be in uninitialized state if we return here. Should we call Shutdown() here?

Reply:
Great catch, fixed